### DISCUSSION TITLE
Inconsistent value in reading xpt

### DISCUSSION THREAD

**Discussion by guoranba** [Roche/pyreadstat#175] [Q&A] [answered]
Hi all,

I run the below 2 codes to read datasets in xpt format and got different values for TRT02PN. It should be NaN for SUBJID 17-004 but it’s 9.380616e-312 when reading ds1.xpt.  

Code 1:
```
ds1tup = prs.read_xport("ds1.xpt",encoding="euc-cn")  
ds1=ds1tup[0].copy()
ds1=ds1.loc[ds1.SUBJID=="17-004"].copy()
ds1.head()
```

 ![ds1](https://user-images.githubusercontent.com/103716748/163514905-4e033b46-ae8d-4948-b5b8-64f6b34fa17b.png)

Code 2:
```
ds2tup = prs.read_xport("ds2.xpt",encoding="euc-cn")  
ds2=ds2tup[0].copy()
ds2=ds2.loc[ds2.SUBJID=="17-004"].copy()
ds2.head()
```

 ![ds2](https://user-images.githubusercontent.com/103716748/163514911-376f2ea0-e443-41e4-8e4d-96aef0b0885e.png)

Both ds1.xpt and ds2.xpt have 3 variables: SUBJID, ASEQ, and TRT02PN. There are 127116 rows in total for ds1.xpt while, for ds2.xpt, it's the subset of ds1 and there are 1334 rows which I only kept the observations with SUBJID in ("01-001","01-076","16-044","17-004").

Does anyone have any ideas of what’s happening? Thanks in advance.
[ds1.zip](https://github.com/Roche/pyreadstat/files/8493826/ds1.zip)
[ds2.zip](https://github.com/Roche/pyreadstat/files/8493827/ds2.zip)

**Comment by ofajardo** [comment, marked as answer]
It is a bug, I created the issue https://github.com/Roche/pyreadstat/issues/177

**Reply by guoranba** [reply]
Thanks ofajardo. Looking forward to the next release.

**Reply by ofajardo** [reply]
Thanks for the very reproducible and nicely written report!

**Reply by ofajardo** [reply]
it should be now fixed in the new release 1.1.5

**Reply by guoranba** [reply]
It works now. Really appreciated!
